using System;

class DivideByZeroExceptionHandling
{
    // static void Main( string[] args )
    // {
    //     try
    //     {
    //         Console.Write( "Enter an integer numerator: " );
    //         int numerator = Convert.ToInt32( Console.ReadLine() );
            
    //         Console.Write( "Enter an integer denominator: " );
    //         int denominator = Convert.ToInt32( Console.ReadLine() );
            
    //         int result = numerator / denominator;            
    //         Console.WriteLine( "\nResult: {0} / {1} = {2}", numerator, denominator, result );

    //         Console.WriteLine("Thank you");
    //     }
    //     catch ( FormatException formatException )
    //     {
    //         Console.WriteLine( "\n" + formatException.Message );
    //         Console.WriteLine("You must enter two integers. Please try again.\n" );
    //     } 
    //     catch ( DivideByZeroException divideByZeroException )
    //     {
    //         Console.WriteLine( "\n" + divideByZeroException.Message );
    //         Console.WriteLine("Zero is an invalid denominator. Please try again.\n" );
    //     }


    //     //General catch clause or General exception
    //     // catch(Exception e)
    //     // {
    //     //     Console.WriteLine(e.Message);
    //     //     Console.WriteLine("Error occured");
    //     // }
    //     // catch
    //     // {
    //     //     Console.WriteLine("Error occured");
    //     // }
    // }
}